docs/code_contribution_guidelines.md

```diff
@@ -319,6 +320,52 @@ Examples of common patterns w.r.t commit structures within the project:
   * If a PR only fixes a trivial issue, such as updating documentation on a
     small scale, fix typos, or any changes that do not modify the code, the
     commit message should end with `[skip ci]` to skip the CI checks.
+    
+## Sign your git commits
+
+When contributing to `lnd` please sign your
+git commits. This is easy to do and will help in assuring the
+integrity of the tree.
+
+### How to sign your commits?
+
+Provide the `-S` flag (or `--gpg-sign`) to git commit when you commit
+your changes, for example
+
+    git commit -m "Commit message" -S
+
+Optionally you can provide a key id after the -S option to sign with a
+specific key.
+
+### What if I forgot?
+
+You can retroactively sign your previous commit using --amend, for example
+
+    git commit -S --amend
+
+If you need to go further back, you can use the interactive rebase
+command with 'edit'. Replace HEAD~3 with the base commit from which
+you want to start.
+
+    git rebase -i HEAD~3
+
+Replace 'pick' by 'edit' for the commit that you want to sign and the
+rebasing will stop after that commit. Then you can amend the commit as
+above. Afterwards, do
+
+    git rebase --continue
+
+As this will rewrite history, you cannot do this when your commit is
+already merged. In that case, too bad, better luck next time.
+
+If you rewrite history for another reason - for example when squashing
+commits - make sure that you re-sign as the signatures will be lost.
```

Comment:
Maybe we can add an example of how to do this. I use this command:

`git rebase --exec 'git commit --amend --no-edit -n -S' -i HEAD~3`